docs/api/python/rnn.md

```diff
@@ -0,0 +1,339 @@
+    future steps, and thus the whole sequence is required.
+```
+
+Dropout and zoneout are effective regularization techniques that can be applied on RNN. `rnn`
+module provides `DropoutCell` and `ZoneoutCell` for regularization on the output and recurrent
+states of RNN. `ZoneoutCell` takes one RNN cell in the constructor, and are unrolled like
+other cells.
+```python
+zoneout_cell = mx.rnn.ZoneoutCell(lstm_cell, zoneout_states=0.5)
+outputs, states = zoneout_cell.unroll(length=sequence_length, \
+                                      inputs=embedded_seq, \
+                                      layout='TNC', \
+                                      merge_outputs=True)
+```
+`DropoutCell` performs dropout on the output of the sequence input. It can be used in a stacked
+multi-layer RNN setting, which we will cover next.
+
+### Multi-layer cells
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    SequentialRNNCell
+    SequentialRNNCell.add
+```
+
+The `SequentialRNNCell` allows stacking multiple layers of RNN cells to improve the expressiveness
+and performance of the model. Cells can be added to a `SequentialRNNCell` in order, from bottom to
+top. When unrolling, the output of a lower-level cell is automatically passed to the next
+higher-level.
+
+```python
+stacked_rnn_cells = mx.rnn.SequentialRNNCell()
+stacked_rnn_cells.add(mx.rnn.BidirectionalCell(
+                          mx.rnn.LSTMCell(num_hidden=50),
+                          mx.rnn.LSTMCell(num_hidden=50)))
+
+# Dropout the output of the bottom layer BidirectionalCell with a retention probability of 0.5.
+stacked_rnn_cells.add(mx.rnn.DropoutCell(0.5))
+
+stacked_rnn_cells.add(mx.rnn.LSTMCell(num_hidden=50))
+outputs, states = stacked_rnn_cells.unroll(length=sequence_length, \
+                                           inputs=embedded_seq, \
+                                           layout='TNC', \
+                                           merge_outputs=True)
+
+# The output of SequentialRNNCell is the same as that of the last layer.
+# In this case 'outputs' is the symbol 'concat6_output' of shape (sequence_length, batch_size, hidden_dim)
+# The states of the SequentialRNNCell is a list of lists, with each list
+# corresponding to the states of each of the added cells respectively.
+```
+
+### Fused RNN cell
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    FusedRNNCell
+    FusedRNNCell.unfuse
+```
+
+The computation of RNN for an input sequence consists of many GEMM and point-wise operations with
+temporal dependencies dependencies. This could make the computation memory-bound especially on GPU,
+resulting in longer wall-time. By combining the computation of many small matrices into that of
+larger ones and streaming the computation whenever possible, the ratio of computation to memory I/O
+can be increased, which results in better performance on GPU. Such optimization technique is called
+"fusing", which
```

Comment:
change to "...technique is called "fusing"." This post .... in greater detail."

Summmary
- break into two sentences
- more details - > greater detail